src/ui/public/courier/data_source/search_source.js

```diff
@@ -288,5 +292,16 @@ export function SearchSourceProvider(Promise, Private, config) {
     }
   };
 
+  SearchSource.prototype.clone = function () {
+    const clone = new SearchSource(this.toString());
+    clone.set('index', this.get('index'));
+    clone.inherits(this.getParent());
+    return clone;
+  };
+
+  SearchSource.prototype.getESQuery = function () {
+    return this._flatten();
```

Comment:
I'm currently using the `body` and `index` properties, perhaps there's more clear naming for this besides `Query` if we need both of these?